### PR TITLE
Make example tests run off of the CLI, tidy testing/

### PR DESCRIPTION
**Pull request by Ekrekr**
This removes dependence on the API part of the CLI, which should be private

Also tidy up project datasets, and `tests/utils` which was weirdly located.

### CODE REVIEW

File: cli/index_test.ts [hunk 1]
```diff
@@ -2,22 +2,17 @@
...
 import { dump as dumpYaml, load as loadYaml } from "js-yaml";
-import * as os from "os";
 import * as path from "path";
 
-import { ChildProcess, execFile } from "child_process";
+import { execFile } from "child_process";
 import { version } from "df/core/version";
 import { dataform } from "df/protos/ts";
-import { suite, test } from "df/testing";
+import { suite, test, getProcessResult, nodePath, npmPath, corePackageTarPath } from "df/testing";
```
Comment (Tuseeq1) [inline cli/index_test.ts:10]: import should be alphabetized. same in other locations.

File: core/main_test.ts [hunk 2]
```diff
@@ -9,9 +9,8 @@ import { decode64, encode64 } from "df/common/protos";
 import { compile } from "df/core/compilers";
 import { version } from "df/core/version";
 import { dataform } from "df/protos/ts";
-import { suite, test } from "df/testing";
+import { suite, test, asPlainObject } from "df/testing";
```
Comment (Tuseeq1) [inline core/main_test.ts:12]: imports should be alphabetized.

**Review by Tuseeq1 — APPROVED** [review]

File: cli/index_test.ts (hunk 1 above)
Comment (Ekrekr) [reply cli/index_test.ts:10]: Let's not bother if the linter won't do it, and it doesn't look there's a drop-in setting

Comment (Ekrekr) [reply cli/index_test.ts:10]: The linter was throwing, I somehow merged it before the tests finished lol - will update in subsequent PR

**Pull request merged**